CDCgov/prime-reportstream
# Fetch Refactor: Daily Data uses React Query implementation

This PR refactors front-end network code to use React Query when loading anything for the Daily Data and Delivery Details pages.

Test Steps:
1. Ensure test suites aren't failing
2. Spin up RS locally and ensure /daily-data and /report-details/:id are loading and calling the correct api endpoints (all of them should say "/api/waters/..." when looking at the URI in the network tab for the designated calls)
3. Ensure code standards are met

## Changes
- Refactors DeliveryHooks to use React Query
- Adds to DeliveriesMockServer for testing
- Adds fixture generators (toying around with ideas on how we can standardize fixtures)
- Moves types around

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
